### PR TITLE
docs: add contributing guide to library

### PR DESCRIPTION
**Pull request by ebk45**
### What does this PR do?

Adds a CONTRIBUTING.md guide to the github-java-client so we can better inform internal and external users on how to contribute to this library.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+- Branch from the master branch and, if needed, rebase to the current master branch before 
+  submitting your pull request. If it doesn't merge cleanly with master you may be asked to rebase 
+  your changes.
+- Commits should be as small as possible while ensuring that each commit is valid independently
+  (i.e. each commit should compile and the tests should pass).
+- Add tests relevant to the fixed bug or new feature. We love to increase our test coverage so any 
+  contributions made to improving that will be very welcomed.
+
+## Coding standards
```
Comment (felix-seifert) [inline CONTRIBUTING.md:31]:
```suggestion
## Coding Standards
```

For staying consistent with the other headlines

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+- This library is modelled after the [GitHub API](https://docs.github.com/en/rest?apiVersion=2022-11-28) and it has been structured to mimic that. 
+  For example, to access the Teams endpoints, you need to instantiate an Organisation Client  
+  and then a Teams Client as [seen here](./src/main/java/com/spotify/github/v3/clients/OrganisationClient.java). This mirrors the nested structure of the API endpoints such as the 
+  [/orgs/{org}/teams/{team_slug}](https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#list-teams) endpoint
+- We operate a monkey see, monkey do approach to this library. We understand that there are some inconsistencies in the library 
+  in terms of how the tests and/or endpoints are written but we, with your help, are working on creating
```
Comment (felix-seifert) [inline CONTRIBUTING.md:38]: "We are working on creating`... Sounds like a word is missing.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  your changes.
+- Commits should be as small as possible while ensuring that each commit is valid independently
+  (i.e. each commit should compile and the tests should pass).
+- Add tests relevant to the fixed bug or new feature. We love to increase our test coverage so any 
+  contributions made to improving that will be very welcomed.
+
+## Coding standards
+
+- This library is modelled after the [GitHub API](https://docs.github.com/en/rest?apiVersion=2022-11-28) and it has been structured to mimic that. 
+  For example, to access the Teams endpoints, you need to instantiate an Organisation Client
```
Comment (felix-seifert) [inline CONTRIBUTING.md:34]: Should we put things like "Organisation Client" in single ticks as we want to refer to the actual code, i.e. `OrganisationClient`?

File: CONTRIBUTING.md (hunk 2 above)
Comment (ebk45) [reply CONTRIBUTING.md:38]: I think I forgot to finish the sentence here 😅

**Review by felix-seifert — APPROVED** [review]
:+1:

**Pull request merged**